audit/repositories/postgres.go

```diff
@@ -0,0 +1,69 @@
+package repositories
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/odpf/salt/audit"
+	"gorm.io/datatypes"
+	"gorm.io/gorm"
```

Comment:
I think having gorm here and grpc interceptor above (in library) would couple the implementation detail.

Reply:
@krtkvrm do you think shield is going to use this implementation of postgres and grpc interceptor as well?

Reply:
+ @rahmatrhd 
we are moving interceptors to pkg in guardian and letting the user set the "getter" fn for actor